### ISSUE TITLE
Questions about migrating from pre-2.0 version.

### ISSUE THREAD

**Issue by LethiferousMoose** [springfox/springfox#2369]
Please take the time to search the repository, if your question has already been asked or answered.

- [x] What version of the library are you using? 2.8.0

What kind of issue is this?

 - [x] Question.

I am attempting to update our product from the last version of swagger-springmvc to the latest version of SpringFox, and cannot seem to easily find the new way performing certain configurations. In the old library there are three functions in particular that are important:
* ```SwaggerSpringMvcPlugin.resourceGroupingStrategy(...)```
* ```SwaggerSpringMvcPlugin.customAnnotationReaders(...)```
* ```SpringSwaggerConfig.swaggerRequestMappingHandlerMappings()```

Based on my digging around your repo and posts online, I am pretty sure that ```SwaggerSpringMvcPlugin.resourceGroupingStrategy(...)``` is now done via implementing ```ApiListingBuilderPlugin```, although at the moment I do not understand how that gets translated into what were once ```ResourceGroup``` objects. 
My second guess is that ```SwaggerSpringMvcPlugin.customAnnotationReaders(...)``` logic can now be performed using a ```OperationBuilderPlugin``` implementation.

I haven't yet been able to test either of these thoughts in much detail because I am getting blocked by my last bullet point. In our system we have dynamic endpoints that do not directly map to a controller instance, instead they are registered with Spring and handled by shared class. In our original implementation we were exposing these to swagger-springmvc by adding ```RequestMappingInfo``` instances mapped to a dummy controller into the ```RequestMappingHandlerMappings``` returned via ```SpringSwaggerConfig.swaggerRequestMappingHandlerMappings()```. 
My question is, is there still a way to register additional ```RequestMappingInfo``` instances in SpringFox that do not directly correlate to an annotated REST controller?

**Comment by dilipkrish** [comment]
@LethiferousMoose Now I need to refresh my memory from 3 yrs ago :) I'll try to answer shortly

**Comment by LethiferousMoose** [comment]
@dilipkrish So I played around with all the plugins, it looks like ```OperationBuilderPlugin``` has the data for the request mappings  I need (i.e. I can see my endpoint names), but there are not accessors to access the ```RequestHandler``` properties on the ```OperationContext``` from embedded the ```RequestMappingContext```. I suppose I could also use reflection to read the data I need from the handler to set on the ```OperationBuilder```.

**Comment by dilipkrish** [comment]
Its not clear what you're trying to do. I'd simply start by replacing the `SwaggerSpringMvcPlugin` with the docket API. and see where that takes you. Do you have customizations using the 1.0 library?

**Comment by dilipkrish** [comment]
@LethiferousMoose No that is not, but it is replaced but much better [extensibility and options](http://springfox.github.io/springfox/docs/current/#plugins)

**Comment by LethiferousMoose** [comment]
@dilipkrish Yeah , I don't think that's going to help, I spent all week playing around with those plugins last week and none of them allow me to read the request mapping info to update the summary, notes, tags, and parameters.

**Comment by dilipkrish** [comment]
😢 the plugin model is very different and much richer in 2.x code line

**Comment by dilas** [comment]
@LethiferousMoose I feel your pain, trying to get the some metadata (present in RequestHandler) to generate any kind of output in my case (eg: Postman Collections, Insomnia data, plain text based documentation, <insert any output that can be used with templates here>, etc).

I just saw a issue #1624 discussing the rationale for hiding this kind of information. Of course, we can add more annotations ourselves to solve this, but it feels redundant IMHO.

Maybe have some kind of technology (spring-mvc, spring-data-rest, grails) agnostic metadata (pojo) hiding implementation details. What do you think @dilipkrish ?

**Comment by dilipkrish** [comment]
@LethiferousMoose Totally agree with hiding the implementation details. That was the reason RequestHandler is introduced.
